src/app/_components/Map/MapMarker.js

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useRef } from 'react';
+import { Marker, Popup } from 'react-leaflet';
+import PropTypes from 'prop-types';
+
+export function MapMarker({ isActive, children, ...props }) {
+  const markerRef = useRef();
+
+  useEffect(() => {
+    const marker = markerRef.current;
+    if (isActive) {
+      marker.openPopup();
+    } else {
+      marker.closePopup();
+    }
+
+    return () => {
+      if (marker) {
+        marker.closePopup();
+      }
```

Comment:
Would `marker?.closePopup()` work?

Reply:
When I tried it, it didn't work, but now it works)